src/main/fc/fc_core.c

```diff
@@ -397,6 +397,16 @@ void processRx(timeUs_t currentTimeUs)
     static bool armedBeeperOn = false;
     static bool airmodeIsActivated;
 
+    if (rxRuntimeConfig.processingRequired) {
+        rxRuntimeConfig.processingRequired = !rxRuntimeConfig.rcProcessFrameFn(&rxRuntimeConfig);
+    }
+
+    if (!rxRuntimeConfig.rxDataReceived) {
+        return;
+    }
+
+    rxRuntimeConfig.rxDataReceived = false;
+
     calculateRxChannelsAndUpdateFailsafe(currentTimeUs);
```

Comment:
Then change `calculateRxChannelsAndUpdateFailsafe` to return `bool` so you can do:
```
if (!calculateRxChannelsAndUpdateFailsafe(currentTimeUs)) {
    return;
}
```